doc/evp_skey.md

```diff
@@ -0,0 +1,99 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represnted as octet buffer. It may
+be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
+
+This is a design of using opaque usage in symmetric encryption.
+
+Key management
+--------------
+
+The provider operating EVP_SKEY objects has to implement a separate key
+management to deal with opaque object. We already have MAC key management and
+it looks suitable for dealing with EVP_SKEY objects.
+
+Allocation and freeing
+--------
+
+```C
+EVP_SKEY * EVP_SKEY_new(void);
+void EVP_SKEY_free(EVP_SKEY *key);
+```
+
+Importing and exporting parameters
+----------------------------------
+
+```C
+int EVP_SKEY_fromdata(EVP_CIPHER_CTX *ctx, EVP_SKEY **pskey, OSSL_PARAM params[]);
+int EVP_SKEY_todata(const EVP_SKEY *key, OSSL_PARAM **params);
+int EVP_SKEY_export(const EVP_SKEY *key, OSSL_PARAM **params,
+                    OSSL_CALLBACK *export_cb, void *export_cbarg);
+```
+
+To find out the parameters supported by a particular EVP_CIPHER object,
+we need a function
+
+```C
+const OSSL_PARAM *EVP_SKEY_fromdata_settable(EVP_CIPHER *cipher);
+```
+
+similar to `EVP_PKEY_fromdata_settable`
+
+API to derive a symmetric key
+-----------------------------
+
+```C
+int EVP_PKEY_derive_SKEY(EVP_PKEY_CTX *ctx, EVP_SKEY *pskey);
```

Comment:
What is this meant to do?  I know of *decrypting* a symmetric key with a public key...